### DISCUSSION TITLE
Object Flip

### DISCUSSION THREAD

**Discussion by warrenvail** [SVG-Edit/svgedit#457] [New feature request]
There are two variations of this, Object Flip Horizontal, and Object Flip Vertical,  What this means is you can copy an object, then selecting the new object, you can flip it horizontally so that it looks like a mirror reflection of the copied object, in other words you can have a drawing of a left shoulder, and flipping it horizontally would make it appear to be a right shoulder.  Flipping an object vertically after copying it would produce a mirror image as seen as a reflection in a lake.  Both of these flips are useful in many ways and would be a great addition to apply to any of the drawing objects, unless of course, you already have that capability and I missed it.

**Comment by dlarue** [comment]
I've found the rotation tool( either the handle or the "angle" text box but not flip or mirror options.
Students are designing airplane wings and tails and symmetry is critically important so designing half a wing, rotating and flipping would be the way to do this.
A flip can be simulated by using move handles along the X axis and moving the right edge past the left edge and being careful to move the exact distance away.  Not the easiest but a possible work-around.

**Comment by jfhenon** [comment]
The toolbar is already a bit busy but this is a good suggestion.

**Reply by dlarue** [reply]
The middle school class is doing great using SVG-edit on their Chromebooks making cardboard airplane gliders. They are using the Path Tool which allows them to make a closed shape which is cut on a laser cutter. I've had to instruct them to use the grid and snap tool to make symmetrical wing and tail sections and some struggle with the concept. I do think that having a mirror option and the ability to attach parts to make a closed loop would be much easier and reinforce the need for horizontal balance in the design.
I have not yet figured out if we can add a path node to an existing design so have been doing this myself for them in Inkscape when an iteration requires a minor tweak.
Otherwise, it's been quite rewarding seeing the students learn about vector editing and files while also learning about basic aviation design elements(balance, center of gravity, center of pressure, control surfaces, wing surface area related to CP and CG, etc ).

**Comment by broady** [comment]
Hi folks. Here's a diff. Happy to send it upstream, or you are welcome to add it yourself if the license is compatible.

https://gist.github.com/broady/1fd55f3b8c65519fc6b7df3b84a02dee

**Reply by dlarue** [reply]
Very nice!  The patch didn't seem to handle the icons correctly or I did something incorrectly.
I had run git pull to update svgedit then ran npm i followed by npm run build --workspace @svgedit/svgcanvas
and verified svgedit ran with npm run start.
Then I applied the patch with git update ../svgedit-flip-patch.diff followed by npm run build --workspace @svgedit/svgcanvas
and then ran it again. The flipping works in both directions but the icons are not set.
<img width="478" height="320" alt="image" src="https://github.com/user-attachments/assets/06a5f85e-a4ad-4a2a-8ce9-fbb0b4a3c311" />

I've created 2 icons from an existing one converting the rectangle to the triangle. Added them to src/editor/images directory and rebuilt the canvas.
<img width="610" height="340" alt="image" src="https://github.com/user-attachments/assets/7d6c4ef1-e641-452a-b2e3-1435f7e8c398" />

![flip_horizontal](https://github.com/user-attachments/assets/c97db418-0478-404a-b726-247fd662ce5f)
![flip_vertical](https://github.com/user-attachments/assets/f2cdf456-38f1-44d9-afb6-cd4047f6693f)

**Reply by broady** [reply]
@dlarue I didn't upload my icons, sorry. Glad the patch helped.

**Comment by jfhenon** [comment]
I will add this to a newer version today - thks

**Comment by jfhenon** [comment]
yes, I will in a later release. Sorry. 
By the way, I used it but added also tests.
